unconstrained_test.go

```diff
@@ -884,16 +915,302 @@ func (Linear) Df(x, grad []float64) {
+		x:       []float64{1, 1},
+		optVal:  0,
+		optLoc:  []float64{1e6, 2e-6},
+		gradTol: 1e-7,
+	},
+	{
+		f:       BrownDennis{},
+		x:       []float64{25, 5, -5, -1},
+		optVal:  85822.20162635628,
+		gradTol: 1e-5, // Use a loose tolerance, otherwise |grad(result.X)| >= settings.GradientAbsTol.
+	},
+	{
+		f:      Trigonometric{},
+		x:      []float64{0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1},
+		optVal: 0,
+		optLoc: []float64{
+			0.0429645656464827, 0.043976286943042, 0.0450933996565844, 0.0463389157816542, 0.0477443839560646,
+			0.0493547352444078, 0.0512373449259557, 0.195209463715277, 0.164977664720328, 0.0601485854398078,
+		},
+	},
+}
+
+var lbfgsTests = []UnconstrainedTest{
+	{
+		f:       Rosenbrock{},
+		x:       []float64{10, 10, 10, 10},
+		optVal:  0,
+		optLoc:  []float64{1, 1, 1, 1},
+		gradTol: 1e-11,
+	},
+	{
+		f:      Rosenbrock{},
+		x:      []float64{-1200000, 1000000},
+		optVal: 0,
+		optLoc: []float64{1, 1},
+	},
+	{
+		f:       Powell{},
+		x:       []float64{0, 1},
+		optVal:  0,
+		optLoc:  []float64{1.09815933e-5, 9.10614674},
+		gradTol: 1e-9,
+	},
+	generateVariablyDimensioned(1000, 1e-10),
+	generateVariablyDimensioned(10000, 1e-8),
+	{
+		f:       Penalty2{},
+		x:       []float64{0.5, 0.5, 0.5, 0.5},
+		optVal:  9.37629e-6,
+		gradTol: 1e-9,
+	},
+	{
+		f:       Penalty2{},
+		x:       []float64{0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5},
+		optVal:  2.93660e-4,
+		gradTol: 1e-9,
+	},
+	{
+		f:       Brown{},
+		x:       []float64{1, 1},
+		optVal:  0,
+		optLoc:  []float64{1e6, 2e-6},
+		gradTol: 1e-9,
+	},
+	{
+		f:       BrownDennis{},
+		x:       []float64{25, 5, -5, -1},
+		optVal:  85822.20162635628,
+		gradTol: 1e-4, // Use a loose tolerance, otherwise |grad(result.X)| >= settings.GradientAbsTol.
+	},
+	{
+		f:      Trigonometric{},
+		x:      []float64{0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1},
+		optVal: 0,
+		optLoc: []float64{
+			0.0429645656464827, 0.043976286943042, 0.0450933996565844, 0.0463389157816542, 0.0477443839560646,
+			0.0493547352444078, 0.0512373449259557, 0.195209463715277, 0.164977664720328, 0.0601485854398078,
+		},
+		gradTol: 1e-9,
+	},
+}
+
+func generateVariablyDimensioned(dim int, gradTol float64) UnconstrainedTest {
+	x := make([]float64, dim)
+	for i := range x {
+		x[i] = float64(dim-i-1) / float64(dim)
+	}
+	optLoc := make([]float64, dim)
+	for i := range optLoc {
+		optLoc[i] = 1
+	}
+	return UnconstrainedTest{
+		f:       VariablyDimensioned{},
+		x:       x,
+		optVal:  0,
+		optLoc:  optLoc,
+		gradTol: gradTol,
+	}
+}
+
 func TestMinimize(t *testing.T) {
-	testMinimize(t, nil)
+	// TODO: When method is nil, Local chooses the method automatically. At
+	// present, it always chooses BFGS (or panics if the function does not
+	// implement Df() or FDf()). For now, run this test with the simplest set
+	// of problems and revisit this later when more methods are added.
+	// Alternatively, we could remove (or better, change) this test because
```

Comment:
No, this test is here to make sure the full integrated test with Local works when the optimizer is unspecified.  We can add TestGetDefaultMethod if we want to check that does the right thing. This test here is a full-integration test, and given how common the call will be, it's worth having even if it is redundant with other tests.